src/autopas/containers/LinkedCells.h

```diff
@@ -112,31 +112,33 @@ class LinkedCells : public ParticleContainer<Particle, ParticleCell, SoAArraysTy
   }
 
   void updateContainer() override {
-    // @todo optimize
-    std::vector<Particle> invalidParticles;
-// custom reduction with templates not supported
-//#pragma omp parallel reduction(vecMerge: invalidParticles)
-#ifdef AUTOPAS_OPENMP
-#pragma omp parallel
-#endif  // AUTOPAS_OPENMP
-    {
-      std::vector<Particle> myInvalidParticles;
-      for (auto iter = this->begin(); iter.isValid(); ++iter) {
-        myInvalidParticles.push_back(*iter);
-      }
+    auto haloIter = this->getRegionIterator(this->_cellBlock.getHaloBoxMin(), this->_cellBlock.getHaloBoxMax(),
```

Comment:
should be done with the normal iterator (+ haloCellsOnly) and not with the RegionIterator.
you will anyways iterate through all halo Cells, which is exactly what the normal iterator will do.

Reply:
fixed 1084179